### DISCUSSION TITLE
adding additional directories to search for exe / binaries, similar to how a shell uses the PATH env var.

### DISCUSSION THREAD

**Discussion by ipatch** [davatorium/rofi#2002] [Q&A] [answered]
hello 👋,

i'm using rofi as a application launcher primarily to spawn gui based applications for my desktop environment. my current setup is a arch linux install, with X11, lightdm, and i3. i came across the below thread in these discussions, tried a couple of things mentioned but can not get additional path entries to work rofi.

https://github.com/davatorium/rofi/discussions/1281#discussioncomment-412050

what i have tried so far, it adding a shell script in the below dir `/etc/profile.d`

https://github.com/ipatch/dotfiles/commit/358dd7b44f3aa367cee2f4b8f1b1313f89ceb1ed

i then run the command suggested in the linked discussion, and i see,

```
cat /proc/$(pidof i3)/environ | tr '\0' '\n' | grep '^PATH'
```

and get the below output

```
PATH=/home/capin/hombrew/sbin:/home/capin/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/opt/rocm/bin:/var/lib/snapd/snap/bin
```

basically i have some gui applicaitons installed `~/homebrew/bin` that i'd like to run and take precedence of a system level package via pacman that is installed in `/usr/bin`

i have logged out and logged back in of lightdm, but have not fully restarted the computer yet, but if that is required i will.

**is there something i'm still missing?**

**Comment by DaveDavenport** [comment]
> PATH=/home/capin/hombrew/sbin:/home/capin/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/opt/rocm/bin:/var/lib/snapd/snap/bin

~why is it **sbin** and **hombrew** in `PATH=/home/capin/hombrew/sbin`  when you want it in `~/homebrew/bin`~
sorry I was trown off and missed  the next entry

this should work.

**Reply by DaveDavenport** [reply, marked as answer]
run 
```bash
G_MESSAGES_DEBUG=Modes.Run rofi -show run -no-config
```

to see what directories it tries to load.

**Reply by ipatch** [reply]
thank you for the helpful debug cmd. 🙏 it sure does indeed so the binary i was looking for. just must've over looked it 🧓.

the output from the debug command you gave me was the below,

```
(process:3076773): Modes.Run-DEBUG: 15:44:18.159: Checking path /opt/code/git/local/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.160: Checking path /home/capin/.rvm/gems/default/wrappers for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.161: Checking path /home/capin/.deno/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.161: Checking path /home/capin/.cargo/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.162: Checking path /home/capin/go/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.163: Checking path /home/capin/.bun/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.163: Checking path /home/capin/.rvm/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.164: Checking path /home/capin/.local/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.168: Checking path /home/capin/homebrew/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.219: Checking path /home/capin/homebrew/sbin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.221: Checking path /usr/local/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.221: Checking path /usr/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.236: Checking path /bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.251: Checking path /usr/local/sbin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.252: Checking path /opt/android-sdk/tools for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.252: Checking path /opt/android-sdk/tools/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.253: Checking path /usr/lib/jvm/default/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.253: Checking path /usr/bin/site_perl for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.254: Checking path /usr/bin/vendor_perl for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.254: Checking path /usr/bin/core_perl for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.254: Checking path /opt/rocm/bin for executable.
(process:3076773): Modes.Run-DEBUG: 15:44:18.255: Checking path /var/lib/snapd/snap/bin for executable.
```

which the same number of path entries for my `SHELL` ie. fish. does rofi some how interpret the `PATH` entries from shell? or am i missing something?

and i should have mentioned i launch rofi with the below key binding using i3.

https://github.com/ipatch/dotfiles/blob/b2a8fccffea76e8df6783449338572a49e60bed6/config/i3/config#L76

**Reply by DaveDavenport** [reply]
rofi inherits the environment from the program that launches it.

**Comment by mostafatouny** [comment]
For the record, you can set the directories rofi searches for executables in by

```
PATH=/usr/local/bin:/usr/bin:/bin:/home/touny/.local/bin rofi -modi run -show run
```

Nothing fancy; We are just launching a command with a custom environment variable. You can set keybindings and desktop entries using it.
